package/kedro_viz/api/graphql/schema.py

```diff
@@ -78,17 +78,21 @@ def run_tracking_data(
         # TODO: this handling of dataset.runs is hacky and should be done by e.g. a
         #  proper query parameter instead of filtering to right run_ids here.
         # Note we keep the order here the same as the queried run_ids.
-        return [
-            TrackingDataset(
-                dataset_name=dataset.dataset_name,
-                dataset_type=dataset.dataset_type,
-                data=format_run_tracking_data(
-                    {run_id: dataset.runs[run_id] for run_id in run_ids},
-                    show_diff,
-                ),
-            )
-            for dataset in tracking_dataset_models
-        ]
+
+        all_tracking_datasets = []
+
+        for dataset in tracking_dataset_models:
+            runs = {run_id: dataset.runs[run_id] for run_id in run_ids}
+            formatted_tracking_data = format_run_tracking_data(runs, show_diff)
+            if len(formatted_tracking_data) > 0:
```

Comment:
```suggestion
            if formatted_tracking_data:
```
(check that works!)

Reply:
Also, is the `self.runs[run_id] = {}` still needed in the `TrackingDatasetModel` exception? I guess it might still be required so that the `-` are put in the right place on the frontend when a metric is missing, but would be good to check. (_Don't_ make it `None` like the `TODO` says though. If it's still required then `{}` is correct and please remove the `TODO`)

Reply:
this doesn't work. it breaks everything so i have kept it